### PR TITLE
[#50] As a user, I can toggle to update my follow status of another user from the article details screen

### LINKED ISSUE #50: As a user, I can toggle to update my follow status of another user from the article details screen
## Why

When the users logged in the application successfully, they can toggle to follow or unfollow another user in the `Article Details` screen. Following a user will show all the articles of this user in the `My Feeds` tab from the `Home` screen.

## Acceptance Criteria
- [ ] Reuse the `Article Details` screen UI layout from #19.
- [ ] Create a follow button with 2 states for the follow button: `normal` and `selected` state. Default the state to `normal` for now.
- [ ] Add the follow button to be right next to author name label with text: `Follow` for `normal` state and `Unfollow` for `selected` state  as shown in the below design.

## Resources

- Sample screen UI layout with the follow button in `normal` state:

Screen Shot 2021-08-09 at 13 15 33 copy 2

- Sample screen UI layout with the follow button in `selected` state:

Screen Shot 2021-08-12 at 11 06 46 copy

### PR DESCRIPTION
**Pull request by markgravity**
Resolved #50 

## What happened

Add Follow Button into Article Detail screen

## Insight

- `FollowButtonStyle` is added to prepare`dark` style in Profile screen (#72 )

## Proof Of Work

| Normal | Selected |
| --- | --- |
| Screen Shot 2021-10-11 at 14 53 50 | Screen Shot 2021-10-11 at 15 14 29 |

### CODE REVIEW

File: NimbleMedium/Sources/Presentation/Views/FollowButton.swift [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+//  NimbleMedium
+//
+//  Created by Mark G on 11/10/2021.
+//
+
+import SwiftUI
+
+enum FollowButtonStyle {
+    case light
+}
```
Comment (minhnimble) [inline NimbleMedium/Sources/Presentation/Views/FollowButton.swift:12]: just curious why do we need to handle only one style `light` here? Shouldn't we handle style with enum when there are more than 2 styles? 🤔

Comment (minhnimble) [reply NimbleMedium/Sources/Presentation/Views/FollowButton.swift:12]: Yeah but I think it is better that we only add the `enum` to handle style when we need to. Now there is only one case and there is no point adding now for this PR 🙏

Comment (suho) [inline NimbleMedium/Sources/Presentation/Views/FollowButton.swift:12]: Let's move this enum into struct FollowButton

```swift
extension FollowButton {

    enum Style {
        
        case light
    }
}
```

File: NimbleMedium/Sources/Presentation/Views/FollowButton.swift [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+        }
+    }
+
+    private var backgroundColor: Color {
+        switch style {
+        case .light:
+            return isSelected ? Color(R.color.lightGray.name) : .clear
+        }
+    }
+
```
Comment (suho) [inline NimbleMedium/Sources/Presentation/Views/FollowButton.swift:33]: Please move these variables into enum Style instead

**Pull request merged**